app/src/main/java/com/app/foodforyou/view/MainActivity.java

```diff
@@ -184,45 +184,65 @@ private void onClickCategory() {
         mStudyDietMainCategory.setOnClickListener(new View.OnClickListener() {
             @Override
             public void onClick(View v) {
-                PreferenceManager.setString(mContext, DIET_SEPARATION_CODE_KEY, mDietSeCode.get(0));
-                PreferenceManager.setString(mContext, MAIN_CATEGORY_NAME_KEY, getString(R.string.study_diet_main_category_name));
-                startActivity(mIntent);
+                if (mDietSeCode.size() > 0) {
```

Comment:
Since this code block is repeated across different functions and the only difference is the dietSeCode value, please refactor by making one method with dietSeCode value as parameter and call that. This will make the code cleaner and also easier to maintain later :)

Reply:
good, I tried to make new method or just set if condition on other methods. I failed many times because of nullPointException.. I think I couldn't catch it how to handle it and where I should put it on which method.
Could you give me hint to progress next step? thank you :)

Reply:
I meant something like this :)

```
private void setCategory(int index, int stringId) {
              if (mDietSeCode.size() > 0) {
                    PreferenceManager.setString(mContext, DIET_SEPARATION_CODE_KEY, mDietSeCode.get(index));
                    PreferenceManager.setString(mContext, MAIN_CATEGORY_NAME_KEY, getString(stringId));
                    startActivity(mIntent);
                } else {
                    showUnstableNetworkErrorDialog();
                }
}
```

then all you have to do is add one line, like this :)

```
        mStudyDietMainCategory.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
            setCategory(0, R.string.study_diet_main_category_name);
        });
```

The general rule is you always have to make sure not to repeat chunks of code so its easier to read and manage.

Reply:
yeah, it looks very good and clear!! What I missed was I focused only ONE parameter not two things. so I failed to refactoring..
Thanks a lot! I've updated it :)